src/base/bittorrent/torrentimpl.cpp

```diff
@@ -463,7 +463,13 @@ qlonglong TorrentImpl::wastedSize() const
 
 QString TorrentImpl::currentTracker() const
 {
-    return QString::fromStdString(m_nativeStatus.current_tracker);
+    if (!m_nativeStatus.current_tracker.empty())
+        return QString::fromStdString(m_nativeStatus.current_tracker);
+
+    if (trackers().size() > 0)
+        return trackers().value(0).url;
```

Comment:
```suggestion
    if (const auto trackers = this->trackers(); !trackers.isEmpty())
        return trackers.constFirst().url;
```

Reply:
@Chocobo1 
What are the objective improvements?

Reply:
* Isn't `isEmpty()` preferred over `size() > 0`? At least in C++ this is the idiom AFAIK.
* Less function call: `trackers` variable, `trackers.constFirst()`
* `QList::value` has boundary checks which is slower.
* Non-const QList methods have a check to an internal atomic variable which is unnecessary when we know it doesn't need to detach.

Reply:
> * Isn't `isEmpty()` preferred over `size() > 0`? At least in C++ this is the idiom AFAIK.

Okay, let's say. Although I usually use it too, but I consider it more a matter of taste, rather than some kind of idiom.
> * Less function call: `trackers` variable, `trackers.constFirst()`

I have never considered the call of a trivial getter as something that needs to be optimized in such a way. But if you insist... 



> * `QList::value` has boundary checks which is slower.

👍 



> Non-const QList methods includes a check to an internal atomic variable which is unnecessary when we know it doesn't need to detach.

How is it applicable to my initial code? Both `size()` and `value()` are `const`.